src/main/java/echo/backend/Parser.java

```
@@ -0,0 +1,187 @@
+        this.statetype = s;
+    }
+    public void keepTemp(String s, int index) {
+        tempStrings[index] = s;
+    }
+    public void resetTemp() {
+        this.tempStrings = new String[]{"", "", "", ""};
+    }
+    /**
+     * Splits an event description from the "/from" keyword to separate the start date.
+     *
+     * @param arg the event description possibly containing the "/from" keyword
+     * @return a String array containing the description and start date
+     */
+    private String[] parseEventFrom(String arg) {
+        return arg.split("/from ");
+    }
+    /**
+     * Splits an event description from the "/to" keyword to separate the end date.
+     *
+     * @param arg the event description possibly containing the "/to" keyword
+     * @return a String array containing the start date and end date
+     */
+    public String[] parseEventTo(String arg) {
+        return arg.split("/to ");
+    }
+    /**
+     * Splits a deadline description from the "/by" keyword to separate the deadline date.
+     *
+     * @param arg the deadline description possibly containing the "/by" keyword
+     * @return a String array containing the description and deadline date
+     */
+    private String[] parseDeadline(String arg) {
+        return arg.split("/by ");
+    }
+    /**
+     * Parses a date string into a LocalDate object based on predefined date formats.
+     *
+     * @param s the date string to be parsed
+     * @return the parsed LocalDate object
+     * @throws DateTimeParseException if the date string cannot be parsed with any of the formats
+     */
+    public LocalDate parseDate(String s) throws DateTimeParseException {
+        DateTimeFormatter formatter;
+        for (String pattern : DATE_FORMATS) {
+            formatter = DateTimeFormatter.ofPattern(pattern);
+            try {
+                LocalDate localDate = LocalDate.parse(s, formatter);
+                return localDate;
+            } catch (DateTimeParseException e) {
```

Consider adding a comment why the handling of this exception is ignored.